### PR TITLE
Add exclude log to agent config

### PR DESCRIPTION
**Pull request by aphralG**
### Proposed changes

Added exclude logs option to agent config 

- Added check where access & error logs are checked against ignorable log types (`syslog`, `off`, `/dev/stderr`, etc) and excluded logs
- If log is outside allowed directories warning is logged that log being read is outside of allowed directory

Fixed typo, changed `TreatWarningsAsError` to `TreatWarningsAsErrors` 

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: internal/watcher/instance/nginx_config_parser.go [hunk 1]
```diff
@@ -146,6 +152,31 @@ func (ncp *NginxConfigParser) createNginxConfigContext(
...
+	ignoreLogs := []string{"off", "/dev/stderr", "/dev/stdout", "/dev/null"}
+
+	if strings.HasPrefix(logLower, "syslog:") || slices.Contains(ignoreLogs, logLower) {
+		return true
+	}
+
+	for _, path := range strings.Split(ncp.agentConfig.DataPlaneConfig.Nginx.ExcludeLogs, ",") {
+		ok, err := filepath.Match(path, logPath)
+		if err != nil {
+			slog.Error("invalid path for excluding log", "log_path", path)
```
Comment (dhurley) [inline internal/watcher/instance/nginx_config_parser.go:166]:
```suggestion
			slog.Error("Invalid path for excluding log", "log_path", path)
```

**Review by dhurley — APPROVED** [review]

**Review by RRashmit — APPROVED** [review]

File: internal/config/testdata/nginx-agent.conf [hunk 2]
```diff
@@ -11,7 +11,8 @@ watchers:
 data_plane_config:
     nginx:
         reload_monitoring_period: 30s
-        treat_warnings_as_error: false
+        treat_warnings_as_errors: false
+        exclude_logs: "/var/log/nginx/error.log,/var/log/nginx/access.log"
```
Comment (oliveromahony) [inline internal/config/testdata/nginx-agent.conf:15]: can we make this consistent with config_dirs ?

Comment (aphralG) [reply internal/config/testdata/nginx-agent.conf:15]: we can, I was just keeping it consistent with what we had in v2 but can change it

Comment (aphralG) [reply internal/config/testdata/nginx-agent.conf]: Changed

**Review by oliveromahony — APPROVED** [review]

**Review by dhurley — APPROVED** [review]

**Review by craigell — APPROVED** [review]

**Review by sean-breen — APPROVED** [review]

**Pull request merged**